### PR TITLE
GSS bug, Verify there is no xss vulnerability with noobaa management console.

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>

### CODE REVIEW

**Review by fbalak — APPROVED** [review]

File: tests/manage/z_cluster/test_noobaa_console.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        url_path = f"https://{url}/robots.txt"
+        html_content = self.get_html_content(url_path)
+        logging.info(f"HTML content:{html_content}")
+        expected_string = ">We dug the earth, but couldn't find your requested URL</"
+        if expected_string not in html_content:
+            ValueError(
+                f"expected_string:{expected_string}, not found on html file:{html_content}"
+            )
+
+    def get_html_content(self, url):
```
Comment (parikshithb) [inline tests/manage/z_cluster/test_noobaa_console.py:48]: Can be moved to helpers.

**Review by parikshithb — COMMENTED** [review]
looks good.

File: tests/manage/z_cluster/test_noobaa_console.py (hunk 1 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_noobaa_console.py:48]: @parikshithb I dont think `get_html_content` is "general function" because the HTTPError.

**Review by parikshithb — APPROVED** [review]

**Review by parikshithb — DISMISSED** [review]

File: tests/manage/z_cluster/test_noobaa_xss_vulnerability.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        url = get_noobaa_url()
+        url_path = f"https://{url}/robots.txt"
+        html_content = self.get_html_content(url_path)
+        logging.info(f"HTML content:{html_content}")
+        expected_string = ">We dug the earth, but couldn't find your requested URL</"
+        if expected_string not in html_content:
+            ValueError(
+                f"expected_string:{expected_string}, not found on html file:{html_content}"
+            )
+
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_noobaa_xss_vulnerability.py:48]: Please also log a success message here.

File: tests/manage/z_cluster/test_noobaa_xss_vulnerability.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        """
+        url = get_noobaa_url()
+        url_path = f"https://{url}/robots.txt"
+        html_content = self.get_html_content(url_path)
+        logging.info(f"HTML content:{html_content}")
+        expected_string = ">We dug the earth, but couldn't find your requested URL</"
+        if expected_string not in html_content:
+            ValueError(
+                f"expected_string:{expected_string}, not found on html file:{html_content}"
+            )
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply tests/manage/z_cluster/test_noobaa_xss_vulnerability.py:47]: done

File: tests/manage/z_cluster/test_noobaa_xss_vulnerability.py (hunk 2 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_noobaa_xss_vulnerability.py:48]: done

**Review by yitzhak12 — DISMISSED** [review]
LGTM

**Review by liranmauda — APPROVED** [review]
LGTM

**Review by PrasadDesala — DISMISSED** [review]

**Review by parikshithb — APPROVED** [review]

**Review by fbalak — APPROVED** [review]

File: ocs_ci/helpers/helpers.py [hunk 4]
```diff
@@ -3272,3 +3272,16 @@ def clear_crash_warning_and_osd_removal_leftovers():
...
+
+def get_noobaa_url():
+    """
+    Get the URL of noobaa console
+
+    Returns:
+        str: url of noobaa console
+
+    """
+    ocp_obj = OCP(kind="routes", namespace=defaults.ROOK_CLUSTER_NAMESPACE)
```
Comment (prsurve) [inline ocs_ci/helpers/helpers.py:3285]: kind `route` can be added to constant

Comment (OdedViner) [reply ocs_ci/helpers/helpers.py]: done

**Review by prsurve — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**